src/sortingtraining/sortalgorythms/BucketSort.java

```diff
@@ -9,42 +9,57 @@
 
         int[] numbsToSort = Arrays.copyOf(numbers, numbers.length);
 
+        if (numbsToSort.length <= 1) {
+            System.out.println("WARN: There is nothing to sort!!! \n");
+            return numbsToSort;
+        }
+
         int largest = Integer.MIN_VALUE;
         int lowest = Integer.MAX_VALUE;
 
         //looking for min and max value
-        for (int n : numbsToSort) {
-            if (n >largest) {
-                largest = n;
+        for (int i : numbsToSort) {
+            if (i >largest) {
+                largest = i;
             }
-            if (n < lowest) {
-                lowest = n;
+            if (i < lowest) {
+                lowest = i;
             }
         }
 
+        boolean negatives = false;
+
+        if (lowest < 0) {
+            negatives = true;
+        }
+
         // if we have negative integers, adding the lowest (abs) one to every record in array, to eliminate negatives
-        if  (lowest < 0) {
+        if  (negatives) {
             lowest = Math.abs(lowest);
-            for (int n = 0; n < numbsToSort.length; n++) {
-                numbsToSort[n] += lowest;
+            for (int j = 0; j < numbsToSort.length; j++) {
```

Comment:
it can be `i` again. The previous `i` variable variable scope is to the end of loop.

Reply:
Okay so professionally when loop is ending you can once again utilize the same variable name? 

I thought that every loop within method should have unique int name, so you can easily distinguish them. 

So i can do it like this, right? :

for (int i = 0; i < numbsToSort.length; i++) {
}
//end of first loop

for (int i = 0; i < x.length; i++) {
}
//end of second loop

for (int i = 0; i < y.length; i++) {
}

And I should use next letters in alphabet only if i have another loops nested in working loop right? 

etc..